src/Uno.Toolkit.RuntimeTests/Tests/StyleExtensionsTest.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Toolkit.RuntimeTests.Helpers;
+using Uno.Toolkit.UI;
+using Uno.UI.RuntimeTests;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Media.Imaging;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests;
+
+[TestClass]
+[RunsOnUIThread]
+internal class StyleExtensionsTest
+{
+	[TestMethod]
+	public async Task StyleExtensionsResourcesPropertyAppliedToStyleTest()
+	{
+		// Arrange
+		var colorBrush = new SolidColorBrush(Windows.UI.Colors.DarkGreen);
+		var testKey = "TestKey";
+
+		var resourceDictionary = new ResourceDictionary
+		{
+			{ testKey, colorBrush }
+		};
+
+		var style = new Style(typeof(Button))
+		{
+			Setters =
+			{
+				new Setter { Property = StyleExtensions.ResourcesProperty, Value = resourceDictionary }
+			}
+		};
+
+		var button = new Button
+		{
+			Style = style
+		};
+
+
+		// Act
+		await UnitTestUIContentHelperEx.SetContentAndWait(button);
+
+		var retrievedResources = style.Setters
+			.OfType<Setter>()
+			.First(s => s.Property == StyleExtensions.ResourcesProperty)
+			.Value as ResourceDictionary;
+
+		// Assert
+		Assert.AreEqual(resourceDictionary, retrievedResources);
+		Assert.AreEqual(button.Resources[testKey], colorBrush);
+
+		// Check if the resource is accessible through the button's style
+		var resourceValue = retrievedResources![testKey] as SolidColorBrush;
+		Assert.IsNotNull(resourceValue);
+		Assert.AreEqual(Windows.UI.Colors.DarkGreen, resourceValue!.Color);
```

Comment:
I'm just wondering if this is too much, should just having: 
Assert.AreEqual(button.Resources[testKey"], colorBrush); be better?

Reply:
Also, should I test if we somehow remove old keys?

Reply:
And I guess I'm going to have to remove colors